### ISSUE TITLE
Centerpoint Nuscene coordinates 

### ISSUE THREAD

**Issue by enesdorukadastec** [open-mmlab/mmdetection3d#2434]
Hi, i train centerpoint model on nuscene dataset. However, before the trainining i want to visualize bounding boxes on point cloud and I created a function on test_nuscene_dataset.py file. In terms of the nuscene dataset coordinats represents  [x,y,z (center of bboxes), size_x, size_y, size_x], but when i visualize bounding boxes I notice [x,y,z]  coordinates is bottom center of bounding boxes. 

      `def test_getitem_with_child_multi_sweeps():
                np.random.seed(0)
                point_cloud_range = [-50, -50, -5, 50, 50, 3]
                file_client_args = dict(backend="disk")
                class_names = [
                    "car".  "truck", "trailer", "bus", "construction_vehicle", "bicycle",
                    "motorcycle", "pedestrian", "traffic_cone", "barrier"
                ]
                pipeline = [
                    dict(
                        type="LoadPointsFromFile",
                        coord_type="LIDAR",
                        load_dim=5,
                        use_dim=5,
                        file_client_args=file_client_args),
                    dict(
                        type="LoadPointsFromMultiSweeps",
                        sweeps_num=2,
                        use_dim=[0,1,2,3,4],
                        file_client_args=file_client_args),
                    dict(
                        type="MultiScaleFlipAug3D",
                        img_scale=(1333, 800),
                        pts_scale_ratio=1,
                        flip=False,
                        transforms=[
                            dict(
                                type="GlobalRotScaleTrans",
                                rot_range=[0, 0],
                                scale_ratio_range=[1., 1.],
                                translation_std=[0, 0, 0]),
                            dict(type="RandomFlip3D"),
                            dict(
                                type="PointsRangeFilter",
                                point_cloud_range=point_cloud_range),
                            dict(
                                type="DefaultFormatBundle3D",
                                class_names=class_names,
                                with_label=False),
                            dict(type="Collect3D", keys=["points"])
                        ])
                ]
      
                nus_dataset = NuScenesDataset(
                    f'{HOME_PATH}/mmdetection3d/data/nuscenes/nuscenes_infos_train.pkl',
                    pipeline,
                    f'{HOME_PATH}/mmdetection3d/data/nuscenes',
                    test_mode=True)
                data = nus_dataset[0]
                assert data['img_metas'][0].data['flip'] is False
                assert data['img_metas'][0].data['pcd_horizontal_flip'] is False
                assert data['points'][0]._data.shape == (32242, 5)
            
                data = nus_dataset[1]
                assert data['img_metas'][0].data['flip'] is False
                assert data['img_metas'][0].data['pcd_horizontal_flip'] is False
                assert data['points'][0]._data.shape == (97352, 5)
            
                gt_bboxes_3d = nus_dataset.get_ann_info(1)['gt_bboxes_3d']
                gt_labels_3d = nus_dataset.get_ann_info(1)['gt_labels_3d']
                print("len bboxes = ", len(gt_bboxes_3d))
                print("Scene labels = ", nus_dataset.get_cat_ids(1))
            
                color_list = [
                    (1,0,0), (0,1,0), (0,0,1), (1,1,0), (1,0,1), (1,1,1),
                    (0,0,0), (1,1,0), (1,0,1), (1,1,1)
                ]
            
                pcd = o3d.geometry.PointCloud()
                pcd.points = o3d.utility.Vector3dVector(data['points'][0].data.detach().cpu().numpy()[:,:3])
            
                boxes = []
                for gt_box_3d, gt_label_3d in zip(gt_bboxes_3d, gt_labels_3d):
                    x = gt_box_3d[0].detach().cpu().numpy()
                    y = gt_box_3d[1].detach().cpu().numpy()
                    z = gt_box_3d[2].detach().cpu().numpy()
            
                    w = gt_box_3d[3].detach().cpu().numpy()
                    l = gt_box_3d[4].detach().cpu().numpy()
                    h = gt_box_3d[5].detach().cpu().numpy()
            
                    yaw = gt_box_3d[6].detach().cpu().numpy()
            
                    boxes.append((x,y,z,w,l,h,yaw))
            
                    print("Box coord = ", gt_box_3d)
                    print("Yaw = ", yaw)
                    print("Calculated z of center = ", z + h/2)
                    if int(gt_label_3d) == -1:
                        print("Label =  Unknown")
                    else:
                        print("Label = ", class_names[gt_label_3d])
                    print('-'*50)
            
                radius = 0.15
                box_sets, circle_sets, axis_sets = [], [], []
                for box_params in boxes:
                    x, y, z, w, l, h, yaw = box_params
                    min_bound = np.array([x - w/2 , y - l/2, z - h/2])
                    max_bound = np.array([x + w/2, y + l/2, z + h/2])
                    aabb = o3d.geometry.AxisAlignedBoundingBox(min_bound, max_bound)
                    aabb.color = color_list[gt_label_3d]
            
                    R = np.eye(3)
                    R[0, 0] = np.cos(yaw)
                    R[0, 1] = -np.sin(yaw)
                    R[1, 0] = np.sin(yaw)
                    R[1, 1] = np.cos(yaw)
                    
                    obb = o3d.geometry.OrientedBoundingBox.create_from_axis_aligned_bounding_box(aabb)
                    obb.rotate(R)
                    obb.color = (1, 0, 0)
                    box_sets.append(obb)
            
                o3d.visualization.draw_geometries([pcd] +  box_sets, point_show_normal=True, width=1280, height=720)`

![image](https://user-images.githubusercontent.com/122875953/232041303-5453357b-c5e6-4633-a38c-531c0ddbf331.png)

i defined min and max bounding box coordinate and the results is shown on the above figure 
 
`   min_bound = np.array([x - w/2 , y - l/2, z - h/2])
    max_bound = np.array([x + w/2, y + l/2, z + h/2])   `

But i changed the min and max bound to;
`   min_bound = np.array([x - w/2 , y - l/2, z ])
    max_bound = np.array([x + w/2, y + l/2, z + h])`
 
i got correct result
![image](https://user-images.githubusercontent.com/122875953/232043038-d549d41c-9d07-4272-ac98-f9cfe7dbddd3.png)

**is this bottom center converted to the center of bounding box during the training. If it is, can you show the code or file name????**

**Comment by JingweiZhang12** [comment]
https://github.com/open-mmlab/mmdetection3d/blob/main/mmdet3d/datasets/nuscenes_dataset.py#L175
https://github.com/open-mmlab/mmdetection3d/blob/main/mmdet3d/structures/bbox_3d/base_box3d.py#L68
